### DISCUSSION TITLE
Securing Configuration Access Point Mode

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#12325] [Q&A] [answered]
I was thinking how I can further secure my tasmota device. I am already using wificonfig = 7 to restrict Tasmota config access point to reset only, that way the device does not randomly offer configuration access point when WiFi routers go missing. Now lets say for some reason the device has gone into wifi configuration mode and is offering the access point for configuration, how to minimize the chance of unauthorized connection to the device? 

1) is it possible to launch "tasmota_XXXXXX-####" access point in hidden mode?
2) Is it possible to rename "tasmota_XXXXXX-####" access point to some thing else without the "tasmota" in the beginning, i.e. (Random_XXXXXX-####") That way it is not an invitation alerting everyone to presence of an open tasmota device in the vicinity?

I know both the measures above will offer only rudimentary security and a determined nuisance maker will still be able to discover and access any open access point, hidden or not.

Regards,

Navi

**Comment by sfromis** [comment, marked as answer]
If you talk about "hidden mode" you are not talking about anything related to "security". You can change naming by compiling your own binary.

**Reply by NsinghP** [reply]
> 
> 
> If you talk about "hidden mode" you are not talking about anything related to "security". You can change naming by compiling your own binary.

I think hidden mode will make it slightly less likely to be accessed by anyone walking by my house, at least  tasmota access point will not pop up on their smartphone as an unsecured access point!. For the second point, of renaming the access point,  I don't see any option in user_config_overide file which I can be modify to compile new .bin file.

**Reply by sfromis** [reply]
`#define PROJECT "fortknox"` will change default naming several places, including AP name. But the best hiddenness is, as Jason2866 suggested, to configure wifi details in the binary.

**Comment by Jason2866** [comment]
If you concerned by this (i dont know why since if you do not find your device you will try as long the device does what you want) compile your own build with compiled in wifi credentials. By doing this the devices never acts as Access Point at any time.
You can configure Tasmota to go never in AP mode once it is configured. Wait endless for AP to connect to.

**Reply by ascillato** [reply]
You can add a password to the AP mode. If you compile by yourself, just add a password to the AP mode.

**Reply by NsinghP** [reply]
> 
> 
> If you concerned by this (i dont know why since if you do not find your device you will try as long the device does what you want) compile your own build with compiled in wifi credentials. By doing this the devices never acts as Access Point at any time.
> You can configure Tasmota to go never in AP mode once it is configured. Wait endless for AP to connect to.

Yes, I see that wifi credentials can be hard coded  and compiled by using config_override file. But if you want a device that gives option to select wifi credentials on factory reset/Reset 1, then I think hard coding the credentials is not a good idea.

**Reply by NsinghP** [reply]
> 
> 
> You can add a password to the AP mode. If you compile by yourself, just add a password to the AP mode.

yes, that I will do,

I am assuming changes to the following:

#define WEB_PASSWORD           ""                //

will also set the password for AP mode?

**Reply by sfromis** [reply]
No
`#define WIFI_AP_PASSPHRASE`

**Reply by NsinghP** [reply]
> No
> `#define WIFI_AP_PASSPHRASE`

That wont work in my case, since I want the freedom to choose any WiFi network and passphrase upon reset.

Edit: I think i was misunderstanding the effects of #define WIFI_AP_PASSPHRASE`. It should work, once passphrase is entered, I can enter new credentials.

**Reply by sfromis** [reply]
That is **exactly** what you get. You wanted a password for AP mode only. I just told you how.

**Reply by NsinghP** [reply]
> 
> 
> That is **exactly** what you get. You wanted a password for AP mode only. I just told you how.

Thanks,

**Reply by NsinghP** [reply]
but if i wanted to change the AP name, then how do I go about doing that. Is it possible to do that with a particular #define line?

**Reply by sfromis** [reply]
See my response in the other sub-thread.

**Reply by NsinghP** [reply]
I was sifting through "my_user_config.h" to see all the #define lines, but I guess that is not the way to go as it does not have all the #define lines that can be used, for instance "#define WIFI_AP_PASSPHRASE" is not there. Where should I go to study all the #define lines that can be possibly used in compiling .bin file?

**Reply by sfromis** [reply]
I already told you, in the other sub-thread above. And `my_user_config.h` contains most of the things you can amend using `#define` (but not all).

**Reply by NsinghP** [reply]
> 
> 
> I already told you, in the other sub-thread above.

#define PROJECT "fortknox"` Lets see if I can figure this one out. :)

Edit: Ok so I get it (sort off). when I "#define PROJECT "Banana" all the places where project name is used will reflect "Banana" and now I will have a "Banana_XXXXXX-####") device AP instead of a  "tasmota_XXXXXX-####" AP. 

Thanks for your time. 

Regards,

Navi

Edit2: Here is my understanding from this thread on how to make a sightly more secure tasmota device :

(1) Configuration AP can not be made to operate in hidden mode. 
(2) #define WEB_PASSWORD       "Custom Password here"    //use this to add password to web server
(3) #define WIFI_AP_PASSPHRASE "Custom Passphrase here" //use this to add passphrase to DEFAULT AP
(4) #define PROJECT            "Custom Name here"      //use this to change DEFAULT AP name, this will also make changes in some other places.

**Reply by NsinghP** [reply]
> You can add a password to the AP mode. If you compile by yourself, just add a password to the AP mode.

FYI, I tested by compiling both with  "#define WEB_PASSWORD " and #define WIFI_AP_PASSPHRASE ". The sequence of events from the time I entered the Wifi Passphrase leads me to believe that setting Web password alone is sufficient to secure the device to an irritating level because when the web password is set and as soon as you connect to the "device access point" it asks for web password before letting you through to configure the wifi credentials screen. 

Once the device acquires IP address it again asks for web password before transferring you to the new ip address and then when on the new IP web server it asks for the web password again. So by my count I had to enter the web password three times before I could access the configuration screen of my new device.  (So for ease of quick setup and comfort of some feel good security,  I think your suggestion to set only Wifi_AP-Passphrase is very sound)

![Screenshot (163)](https://user-images.githubusercontent.com/69470381/121712373-5106ce00-caf9-11eb-9d23-52deece5edc4.png)
![Screenshot (164)](https://user-images.githubusercontent.com/69470381/121712421-5b28cc80-caf9-11eb-9868-a4a8d265b2b9.png)
![Screenshot (165)](https://user-images.githubusercontent.com/69470381/121712454-624fda80-caf9-11eb-8f63-ba87234a6b0a.png)
![Screenshot (166)](https://user-images.githubusercontent.com/69470381/121712490-6a0f7f00-caf9-11eb-8cb0-1223cdf48bea.png)

and then one more time at the new ip address.
